python/setup.py

```diff
@@ -90,14 +90,14 @@ def run(self):
     copy2(SCRIPT_PATH, SCRIPT_TARGET)
 
     install_requires = [
-        "numpy",
-        "typing",
-        "pandas == 1.1.4",
-        "psutil",
-        "pyarrow >= 0.10",
-        "ray == 1.4.0",
-        "pyspark >= 3.0.0, < 3.1.0",
-        "netifaces"
+        "numpy < 2.0.0",
+        "typing < 4.0.0",
+        "pandas >= 1.1.4, < 2.0.0",
+        "psutil < 6.0.0",
+        "pyarrow >= 0.10, < 5.0.0",
+        "ray >= 1.4.0, < 2.0.0",
+        "pyspark >= 3.0.0, < 4.0.0",
```

Comment:
Hi @kira-lin, does the raydp support spark 3.1.0?

Reply:
it works fine when building with the older version in the pom.xml:  https://github.com/oap-project/raydp/issues/121#issuecomment-811790060.  need to address that issue separately but this unblocks usage of 3.1.0 which is the minimum required for, e.g., delta-spark (https://pypi.org/project/delta-spark/), which i'd like to integrate

Reply:
Well, if there are demand for spark 3.1, we should give it a try. @mjschock can you please also modify the dependencies in the raydp.yaml to see if it pass CI?
By the way, I actually have the patch to support spark 3.1, but we are in a discussion about whether we should add a shim layer for different spark versions.

Reply:
@kira-lin i updated raydp.yml to use both pyspark 3.0.0 and 3.1.2 and they both pass